File: app/src/main/java/ezlife/movil/oneparkingapp/activities/PassActivity.java
package ezlife.movil.oneparkingapp.activities;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import ezlife.movil.oneparkingapp.R;
import ezlife.movil.oneparkingapp.databinding.PassBinding;
import ezlife.movil.oneparkingapp.net.api.login.LoginApi;
import ezlife.movil.oneparkingapp.util.CryptoJs;
import ezlife.movil.oneparkingapp.util.Qk;
import ezlife.movil.oneparkingapp.util.Settings;

public class PassActivity extends AppCompatActivity implements CryptoJs.OnEncryptAES, LoginApi.OnPassListener {

    PassBinding binding;
    LoginApi api;
    CryptoJs cryptoJs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_pass);

        binding.setHandler(this);
        api = new LoginApi(this, Qk.makeLoading(this));
        cryptoJs = new CryptoJs(this);
    }

    public void changePass(){
        String pass1 = Qk.getText(binding.pass1);
        String pass2 = Qk.getText(binding.pass2);

        if(pass1 == null || pass2 == null || !pass1.equals(pass2)){
            Qk.showToast(this,R.string.pass_error );
        }else{
            String pass = pass1+"_&&_"+ System.currentTimeMillis();
            cryptoJs.encryptAES(pass, Settings.SECRET, this);
        }
    }

    @Override
    public void onEncryptedAES(String aes) {
        api.changePassword(aes, this);
    }

    @Override
    public void onPass(boolean success, int error) {
        if(success){
            Intent intent = new Intent(this, MapActivity.class);
            startActivity(intent);
        }else{
            Qk.showToast(this,R.string.pass_error_2 );
        }
    }

}

File: app/src/main/java/ezlife/movil/oneparkingapp/activities/PassActivity.kt
package ezlife.movil.oneparkingapp.activities

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import ezlife.movil.oneparkingapp.R
import ezlife.movil.oneparkingapp.databinding.PassBinding
import ezlife.movil.oneparkingapp.providers.UserProvider
import ezlife.movil.oneparkingapp.util.Preference
import ezlife.movil.oneparkingapp.util.text

class PassActivity : AppCompatActivity() {

    companion object {
        @JvmStatic val EXTRA_CARS = "cars"
    }

    lateinit var binding: PassBinding
    val provider: UserProvider by lazy { UserProvider(this, makeLoading()) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_pass)
        binding.handler = this
    }

    fun updatePass() {
        val pass1 = "${binding.pass1.text()}"
        val pass2 = "${binding.pass2.text()}"

        if (pass1 != pass2) {
            toast(R.string.pass_error)
            return
        }
        provider.updatePassword(pass1) { success ->
            if (success) {
                val cars = intent.extras.getBoolean(EXTRA_CARS, false)
                if(cars) {
                    savePreference(Preference.USER_LOGGED to true)
                    startActivity<MapActivity>()
                }else{
                    startActivity<AddCarActivity>()
                }
            } else {
                toast(R.string.pass_error_2)
            }
        }
    }
}